assignments/asgmt6.rb

```diff
@@ -0,0 +1,86 @@
+ require 'minitest/autorun'
+  
+  class PriorityQueue
+
+    class SubQueue
+      attr :rank, :items
+      def initialize(rank)
+         @rank = rank
+         @items = []
+      end
+      def enqueue(item)
+        @items<<item
+      end
+      def dequeue
+        @items.shift
+      end
+      def empty?
+        @items.empty?
+      end
+      def peek
+        @items[0]
+      end
+      def length
+        @items.length
+      end
+    end #SubQueue
+    
+    def initialize
+      @high_sq = SubQueue.new(:high)
```

Comment:
I would suggest a slightly different implementation (since this would simplify your enqeuue method). Create a hash with priority as the key:
`@queues = {high: SubQueue.new, med: SubQueue.new, low: SubQueue.new}`